### PR TITLE
WIP - Adding the build definition.

### PR DESCRIPTION
**Pull request by rockydevnet**
Starting to work to defined the CI Build.

### CODE REVIEW

**Comment by rockydevnet** [comment]
The build is currently failing due to a problem running the `Apply version numbers to assemblies` PowerShell. If @elduddz  or @MarcDenman  have any suggestions, they would be welcome!

**Comment by elduddz** [comment]
It says in the logs that it is missing a mandatory variable.

If you click on the details above it will have an error link to the build. Can you get to the error step?

**Comment by elduddz** [comment]
2019-02-05T20:14:19.8582023Z Formatted command: . 'E:\VSTSUnifiedAgents\07-Eric-A01\_work\27\s\DeploymentScripts\Apply-AssemblyVersionAndDefaults.ps1' -buildNumber "UKHO.ConfigurableStub_2019036.14" -solutionDirectory "E:\VSTSUnifiedAgents\07-Eric-A01\_work\27\s"

**Comment by elduddz** [comment]
Could you put some more logging in the script to see how far it is getting? Not sure I can see the variable it says it's missing Process?

**Comment by rockydevnet** [comment]
@elduddz I'm not sure it's actually getting as far as running the script? I've added some Write-Host lines as the first lines of the script, but they don't appear.

Also, the param it's complaining about is `Process`: `Cannot process command because of one or more missing mandatory parameters: Process.`

**Comment by MarcDenman** [comment]
```powershell
2019-02-05T20:58:38.8672544Z Setting assembly versions and defaults...
2019-02-05T20:58:38.8688162Z     BuildNumber: UKHO.ConfigurableStub_2019036.17
2019-02-05T20:58:38.8692515Z     SolutionDir: E:\VSTSUnifiedAgents\HTFSMGTUAG07-Eric-A01\_work\27\s
``` 
It is hitting the write-hosts aren't they? Or am I looking at the wrong thing?

**Comment by rockydevnet** [comment]
@MarcDenman, you are indeed correct! Great... now to binary search for the missing parameter...

**Comment by elduddz** [comment]
Line 22 where does $majorMinorVersion come from?

**Comment by elduddz** [comment]
Ah line 10

**Comment by elduddz** [comment]
From what I can work out on my mobile phone, it isn't getting to line 34, I am wondering if the array cannot be `Processed`

**Comment by rockydevnet** [comment]
When run locally, it was complaining about missing process for the ForEach-Object. I'd mistakenly put the `{` on a new line, which upset it. It looks a little happier now!

**Comment by elduddz** [comment]
Had this created the build called pipeline-ci? What happens with the next one?

**Comment by elduddz** [comment]
why are you force pushing so much?

**Comment by rockydevnet** [comment]
The force push are when I've re-written the previous commit with a minor change to keep history tidy, usually because we can't execute the build locally, so have to commit to see if it builds... The amended commit generally contains a small syntax correction

**Comment by MarcDenman** [comment]
> Last job is to add the LicenceURL to the NuGet package, and push the new NuGet package to ProGet...

[ADM](https://github.com/UKHO/ADM-PSmodule/blob/master/azure-pipelines.yml) is pushing to ProGet so you should just be able to copy the script/step over. It does require variables being set on the build itself(not in the YML) though because of secrets.

```
- powershell: | 
    ./Deploy/DeployToFeed.ps1 -RepositoryName $(RepositoryName) -RepositorySourceUri $(RepositorySourceUri) -RepositoryPublishUri $(RepositoryPublishUri) -NugetAPIKey $(NugetApiKey) -ModuleFolderPath $(Build.SourcesDirectory)\UKHO.ADM
  displayName: 'Publish module to feed'
```

File: azure-pipelines.yml [hunk 1]
```diff
@@ -9,3 +11,27 @@ trigger:
 
 queue:
   name: NautilusBuild
+
+variables:
+  buildConfiguration: 'Release'
+
+steps:
+- task: DotNetCoreInstaller@0
```
Comment (elduddz) [inline azure-pipelines.yml]: looks good for build requirement

**Pull request merged**